src/components/Tables/DataTable/fragments/inputs/TableInputMask.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <InputMask
+    :mask="preset ? presets[preset.toLowerCase()].mask : undefined"
+    :placeholder="
+      preset && edit
+        ? presets[preset.toLowerCase()].mask?.replace(/9/g, '_')
+        : undefined
+    "
+    unmask
+    :readonly="!edit"
+    :class="edit ? '' : '!border-none !shadow-none !bg-transparent'"
+  />
+</template>
+
+<script setup lang="ts">
+import InputMask, { type InputMaskProps } from "primevue/inputmask";
+
+export type TableInputMaskProps = {
+  edit: boolean;
+  preset?: null | string;
+};
+defineProps<TableInputMaskProps>();
+
+type presetsType = { [key: string]: InputMaskProps };
+const presets = <presetsType>{
```

Comment:
Dê preferência ao uso do tipo genérico (comentário geral da review). Além disso, corrija a case e tipe explicitamente:

```suggestion
type PresetsType = { [key: string]: InputMaskProps };
const presets: PresetsType = {
```

Além disso, aqui ocorre o mesmo que no `TableInputBoolean`, linha 16. Ao tiparmos corretamente essa key, tornamos o acesso a lista de presets mais seguro e removemos a necessidade de algumas modelagens (como a que ocorre nas linhas 3 e 6).